Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.cpp

```diff
@@ -488,19 +488,89 @@ void InbandTextTrackPrivateAVF::resetCueValues()
 
 void InbandTextTrackPrivateAVF::setMode(InbandTextTrackPrivate::Mode newMode)
 {
-    if (!m_owner)
-        return;
-
     InbandTextTrackPrivate::Mode oldMode = mode();
     InbandTextTrackPrivate::setMode(newMode);
 
-    if (oldMode == newMode)
+    if (m_owner && oldMode != newMode)
+        m_owner->trackModeChanged();
+}
+
+bool InbandTextTrackPrivateAVF::processVTTFileHeader(CMFormatDescriptionRef formatDescription)
+{
+    using namespace PAL;
+    ASSERT(!m_haveReportedVTTHeader);
+    ASSERT(formatDescription);
+
+    RefPtr<ArrayBuffer> buffer;
+
+    CFDictionaryRef extensions = PAL::CMFormatDescriptionGetExtensions(formatDescription);
+    if (!extensions)
+        return false;
+
+    CFDictionaryRef sampleDescriptionExtensions = static_cast<CFDictionaryRef>(CFDictionaryGetValue(extensions, kCMFormatDescriptionExtension_SampleDescriptionExtensionAtoms));
+    if (!sampleDescriptionExtensions)
+        return false;
+
+    CFDataRef webvttHeaderData = static_cast<CFDataRef>(CFDictionaryGetValue(sampleDescriptionExtensions, CFSTR("vttC")));
+    if (!webvttHeaderData)
+        return false;
+
+    unsigned length = CFDataGetLength(webvttHeaderData);
+    if (!length)
+        return false;
+
+    // A WebVTT header is terminated by "One or more WebVTT line terminators" so append two line feeds to make sure the parser
+    // reccognizes this string as a full header.
+    auto header = makeString(std::span { CFDataGetBytePtr(webvttHeaderData), length }, "\n\n"_s);
+
+    INFO_LOG(LOGIDENTIFIER, "VTT header ", header);
+    notifyMainThreadClient([&](auto& client) {
```

Comment:
I'm not familiar with `notifyMainThreadClient()` but I assume it dispatches an asynchronous task (maybe even to another thread). Either way, if so, capturing header by reference doesn't seem safe and we'd likely want to capture `header = WTFMove(header)` (in no multi-threading) or `header = crossThreadCopy(WTFMove(header))` (if multithreaded).

Reply:
Excellent point. I will fix this, the one you mention below, and the other places in this file that don't show up because I didn't modify them.